inference-engine/include/ie_plugin_config.hpp

```diff
@@ -148,6 +151,36 @@ DECLARE_METRIC_KEY(NUMBER_OF_EXEC_INFER_REQUESTS, unsigned int);
  */
 DECLARE_METRIC_KEY(DEVICE_ARCHITECTURE, std::string);
 
+/**
+ * @brief Enum to define possible device types
+ */
+enum class DeviceType {
+    integrated = 0,
+    discrete = 1,
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const InferenceEngine::Metrics::DeviceType& deviceType) {
+    switch (deviceType) {
+        case InferenceEngine::Metrics::DeviceType::discrete: os << "discrete"; break;
+        case InferenceEngine::Metrics::DeviceType::integrated: os << "integrated"; break;
+        default: os << "unknown"; break;
+    }
+
+    return os;
+}
+/** @endcond */
+
+/**
+ * @brief Metric to get a type of device. See DeviceType enum definition for possible return values
+ */
+DECLARE_METRIC_KEY(DEVICE_TYPE, DeviceType);
```

Comment:
can we add common tests like https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/shared/include/behavior/core_integration.hpp#L759-L773 ?

Reply:
Added